commit c914c89927c12e7b7b67c090b892a071ba2ed2e4 (refs/changes/30/730/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-18T12:41:36-07:00 (3 years ago)
    
    RELENG-729 purge-mg-builds is failing (pipefail) on a grep with no hits
    Reviewed by: Brian Bennett <brian.bennett@joyent.com>

diff --git a/tools/purge-mg-builds b/tools/purge-mg-builds
index 0a6a5f9..55db846 100755
--- a/tools/purge-mg-builds
+++ b/tools/purge-mg-builds
@@ -299,7 +299,7 @@ function purge_mg_builds
     branches=$(echo "$dirs" | sed -E 's/-[0-9]{8}T[0-9]{6}Z$//' | sort | uniq)
 
     for branch in $branches; do
-        branch_dirs=$(echo "$dirs" | grep "^$branch-")
+        branch_dirs=$(echo "$dirs" | (grep "^$branch-" || true))
 
         # Determine which dirs for this branch that we will be purging.
         branch_dirs_to_purge=
@@ -313,12 +313,12 @@ function purge_mg_builds
 
         # The "master" branch is special: if we would end up purging *all*
         # remaining master branch dirs, then save that latest one.
+        purge_all_in_branch=no
         if [[ "$branch_dirs" == "$branch_dirs_to_purge" ]]; then
             if [[ "$branch" == "master" ]]; then
                 log "#   retain last master branch build:" \
                     $(echo "$branch_dirs_to_purge" | tail -1)
                 branch_dirs_to_purge=$(echo "$branch_dirs_to_purge" | sed '$ d')
-                purge_all_in_branch=no
             else
                 purge_all_in_branch=yes
             fi
